### PR TITLE
Fixes #22117 - Add support for number of sockets for ovirt

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

**Comment by shiramax** [comment]
This needs : https://github.com/fog/fog-ovirt/pull/16 to be merged as well.

**Review by ekohl — CHANGES_REQUESTED** [review]
It'll also need to bump the required version in `bundler.d/ovirt.rb` when the library has released a version that supports sockets.

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 1]
```diff
@@ -94,7 +94,11 @@ def provided_attributes
 
     #FIXME
```
Comment (tbrisker) [inline app/models/compute_resources/foreman/model/ovirt.rb]: Can we remove this FIXME? is 16 the max count supported for these by ovirt?

Comment (shiramax) [reply app/models/compute_resources/foreman/model/ovirt.rb]: @tbrisker 
After looking at ovirt engine,  I saw the following:
[root@compute-ge-6 ~]#  engine-config --get MaxNumOfVmCpus
MaxNumOfVmCpus: 240 version: 3.6
MaxNumOfVmCpus: 240 version: 4.0
MaxNumOfVmCpus: 288 version: 4.1
MaxNumOfVmCpus: 384 version: 4.2

[root@compute-ge-6 ~]# engine-config --get  MaxNumOfCpuPerSocket
MaxNumOfCpuPerSocket: 16 version: 3.6
MaxNumOfCpuPerSocket: 16 version: 4.0
MaxNumOfCpuPerSocket: 254 version: 4.1
MaxNumOfCpuPerSocket: 254 version: 4.2

that means the max_socket=16 and max_cores=16 is true to ovirt 3.6 and 4.0 but not to versions >= 4.0 
After taking to @orrabin , we think that at the moment we don't see any reason to set the max socket and cores to be more then 16.
Until now will allowed only 8 cores and after this patch, we will allow 16*16 = 256 cores, which seems to be enough at the moment.

**Review by tbrisker — DISMISSED** [review]
Ack, makes sense. Thanks @shiramax !

**Comment by orrabin** [comment]
I tested that when selecting a template the right number of sockets is filled in, I changed that number and save to see the new number was saved in the host (checked in foreman and ovirt).

**Comment by orrabin** [comment]
@ekohl I added: https://github.com/theforeman/foreman-packaging/pull/2650

@tbrisker - any other comments here?

**Review by tbrisker — APPROVED** [review]
LGTM

**Comment by tbrisker** [comment]
Packaging side should be merged first.

**Comment by shiramax** [comment]
@ekohl, thanks, the packaging PR was merged, is there anything else?

**Review by ekohl — APPROVED** [review]

**Comment by orrabin** [comment]
Thanks @shiramax !

**Pull request merged**